### PR TITLE
Refactored StartDebug_term and EndDebug functions.

### PR DESCRIPTION
**Pull request by ubaldot**
* Functions are way too long. Readability and maintainability should be slightly improved.
* Some variables are re-assigned to their initial value at teardown. This should not be needed since all internal variables are re-initialized at startup of every Termdebug session.

### CODE REVIEW

File: runtime/pack/dist/opt/termdebug/plugin/termdebug.vim [hunk 1]
```diff
@@ -1306,12 +1295,8 @@ def DeleteCommands()
   endif
 
   sign_unplace('TermDebug')
-  breakpoints = {}
-  breakpoint_locations = {}
-
```
Comment (chrisbra) [inline runtime/pack/dist/opt/termdebug/plugin/termdebug.vim]: Those should be kept, no?

File: runtime/pack/dist/opt/termdebug/plugin/termdebug.vim [hunk 2]
```diff
@@ -1306,12 +1295,8 @@ def DeleteCommands()
...
   sign_undefine('debugPC')
   sign_undefine(BreakpointSigns->map("'debugBreakpoint' .. v:val"))
-  BreakpointSigns = []
```
Comment (chrisbra) [inline runtime/pack/dist/opt/termdebug/plugin/termdebug.vim:1314]: That should be kept, no?

File: runtime/pack/dist/opt/termdebug/plugin/termdebug.vim [hunk 3]
```diff
@@ -607,15 +621,16 @@ def StartDebug_prompt(dict: dict<any>)
 
   ch_log($'executing "{join(gdb_cmd)}"')
   gdbjob = job_start(gdb_cmd, {
-    exit_cb: function('EndPromptDebug'),
+    # exit_cb: function('EndPromptDebug'),
```
Comment (chrisbra) [inline runtime/pack/dist/opt/termdebug/plugin/termdebug.vim]: can be removed

**Comment by chrisbra** [comment]
Seems fine for me

File: runtime/pack/dist/opt/termdebug/plugin/termdebug.vim (hunk 1 above)
Comment (ubaldot) [reply runtime/pack/dist/opt/termdebug/plugin/termdebug.vim]: I am not 100% sure, I may be missing something. :) What is the problem in leaving few script variable set when they will be reset anyway during the startup of the next Termdebug session? Do we really need to reset them to empty values at teardown? Why only these vars and not ALL the vars? (I am sure I am missing something though).

Comment (ubaldot) [reply runtime/pack/dist/opt/termdebug/plugin/termdebug.vim]: However, I re-added them.

Comment (chrisbra) [reply runtime/pack/dist/opt/termdebug/plugin/termdebug.vim]: True, I guess we don't neccessarily need this.

**Pull request closed without merging**